taskt/Core/Script/Script.cs

```diff
@@ -1,180 +1,177 @@
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//Unless required by applicable law or agreed to in writing, software
+//distributed under the License is distributed on an "AS IS" BASIS,
+//WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//See the License for the specific language governing permissions and
+//limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Windows.Forms;
+using Newtonsoft.Json;
+using taskt.Core.Automation.Commands;
+
+namespace taskt.Core.Script
+{
+    using Formatting = Newtonsoft.Json.Formatting;
+
+    public class Script
+    {
+        public string FileName { get; set; }
+        public string ProjectName { get; set; }
+        /// <summary>
+        /// Contains user-defined variables
+        /// </summary>
+        public List<ScriptVariable> Variables { get; set; }
+        /// <summary>
+        /// Contains user-selected commands
+        /// </summary>
+        public List<ScriptAction> Commands;
+
+        public Script()
+        {
+            Variables = new List<ScriptVariable>();
+            Commands = new List<ScriptAction>();
+        }
+
+        /// <summary>
+        /// Returns a new 'Top-Level' command.
+        /// </summary>
+        public ScriptAction AddNewParentCommand(ScriptCommand scriptCommand)
+        {
+            ScriptAction newExecutionCommand = new ScriptAction() { ScriptCommand = scriptCommand };
+            Commands.Add(newExecutionCommand);
+            return newExecutionCommand;
+        }
+
+        /// <summary>
+        /// Converts and serializes the user-defined commands into an JSON file
+        /// </summary>
+        public static Script SerializeScript(
+            ListView.ListViewItemCollection scriptCommands,
+            List<ScriptVariable> scriptVariables,
+            string scriptFilePath = "",
+            string projectName = ""
+            )
+        {
+            var script = new Script();
+
+            script.FileName = System.IO.Path.GetFileName(scriptFilePath);
+
+            script.ProjectName = projectName;
+
+            //save variables to file
+            script.Variables = scriptVariables;
+
+            //save listview tags to command list
+            int lineNumber = 1;
+            List<ScriptAction> subCommands = new List<ScriptAction>();
+
+            foreach (ListViewItem commandItem in scriptCommands)
+            {
+                var command = (ScriptCommand)commandItem.Tag;
+                command.LineNumber = lineNumber;
+
+                if ((command is LoopNumberOfTimesCommand) || (command is LoopContinuouslyCommand) ||
+                    (command is LoopListCommand) || (command is BeginIfCommand) ||
+                    (command is BeginMultiIfCommand) || (command is TryCommand) ||
+                    (command is BeginLoopCommand) || (command is BeginMultiLoopCommand))
+                {
```

Comment:
We may want to look at building a loop base class or identifier at some point to allow for more flexibility here.